### PR TITLE
Remove pluggable_node_access records.

### PR DESCRIPTION
**Pull request by HelenaEksler**
#20

### CODE REVIEW

**Comment by HelenaEksler** [comment]
@amitaibu 
Can you review this, please?

File: pluggable_node_access.module [hunk 1]
```diff
@@ -245,3 +245,28 @@ function pluggable_node_access_og_access_invoke_node_access_acquire_grants_alter
...
+ * Implements hook_node_delete().
+ */
+function pluggable_node_access_node_delete($node) {
+  $wrapper = entity_metadata_wrapper('node', $node);
+  if (!isset($wrapper->pluggable_node_access)) {
+    // Node doesn't have restricted access.
+    return;
+  }
+  // Remove pluggable_node_access record when node is deleted.
+  $pluggable_node_access_entities = $wrapper->pluggable_node_access->value();
```
Comment (HelenaEksler) [inline pluggable_node_access.module]: don't hardcode the field name.

``` php
$ids = $wrapper->pluggable_node_access->value(array('dientifier' => TRUE);
$ids = is_array($ids) ? $ids : array($ids);
entity_delete_multiple('pluggable_node_access', $ids)
```

**Comment by HelenaEksler** [comment]
@amitaibu 
Not using the hardcoded field name.

File: pluggable_node_access.module [hunk 2]
```diff
@@ -245,3 +245,27 @@ function pluggable_node_access_og_access_invoke_node_access_acquire_grants_alter
...
   }
 }
+
+/**
+ * Implements hook_node_delete().
+ */
+function pluggable_node_access_node_delete($node) {
+  $wrapper = entity_metadata_wrapper('node', $node);
+  // Get the field name.
+  /** @var PluggableNodeAccessInterface $object */
```
Comment (amitaibu) [inline pluggable_node_access.module]: remove.

File: pluggable_node_access.module [hunk 3]
```diff
@@ -245,3 +245,27 @@ function pluggable_node_access_og_access_invoke_node_access_acquire_grants_alter
...
+  $object = pluggable_node_access_get_pluggable_node_access_handler('email_domain', $node);
```
Comment (amitaibu) [inline pluggable_node_access.module]: $object => $handler.

Anyway, this looks wrong, as it hardcodes to the `email_domain`. I would expect it to work on all plugable_node_access.

File: pluggable_node_access.module [hunk 4]
```diff
@@ -245,3 +245,28 @@ function pluggable_node_access_og_access_invoke_node_access_acquire_grants_alter
...
   }
 }
+
+/**
+ * Implements hook_node_delete().
+ */
+function pluggable_node_access_node_delete($node) {
+  $wrapper = entity_metadata_wrapper('node', $node);
+  foreach(pluggable_node_access_get_pluggable_node_access_plugins() as $plugin_name => $plugin) {
+    // Get fields names
```
Comment (HelenaEksler) [inline pluggable_node_access.module]: missing dot.

File: pluggable_node_access.module [hunk 5]
```diff
@@ -245,3 +245,28 @@ function pluggable_node_access_og_access_invoke_node_access_acquire_grants_alter
     }
   }
 }
+
+/**
+ * Implements hook_node_delete().
+ */
```
Comment (HelenaEksler) [inline pluggable_node_access.module:253]: add comment what we do here.

File: pluggable_node_access.module [hunk 6]
```diff
@@ -245,3 +245,28 @@ function pluggable_node_access_og_access_invoke_node_access_acquire_grants_alter
...
+function pluggable_node_access_node_delete($node) {
+  $wrapper = entity_metadata_wrapper('node', $node);
+  foreach(pluggable_node_access_get_pluggable_node_access_plugins() as $plugin_name => $plugin) {
+    // Get fields names
+    $handler = pluggable_node_access_get_pluggable_node_access_handler($plugin_name, $node);
+    $fields = $handler->getReferenceFields($node);
+
+    foreach($fields as $field) {
+      if (!isset($wrapper->$field)) {
```
Comment (HelenaEksler) [inline pluggable_node_access.module]: `(!$wrapper->{$field}->value()`

**Comment by HelenaEksler** [comment]
add test

**Comment by HelenaEksler** [comment]
@amitaibu Test is green.  Can you review it, please?

**Comment by HelenaEksler** [comment]
Merged manually.

**Pull request closed without merging**